rust/foxglove/src/websocket/service/response.rs

```diff
@@ -45,11 +39,40 @@ impl Responder {
     /// [`ServiceSchema`][super::ServiceSchema]. If no response encoding was declared, then the
     /// encoding is presumed to be the same as the request.
     pub fn set_encoding(&mut self, encoding: impl Into<String>) {
-        self.encoding = encoding.into();
+        if let Some(inner) = self.0.as_mut() {
+            inner.encoding = encoding.into();
+        }
     }
 
     /// Completes the request by sending a response to the client.
-    pub fn respond(self, result: Result<Bytes, String>) {
+    pub fn respond(mut self, result: Result<Bytes, String>) {
+        if let Some(inner) = self.0.take() {
+            inner.respond(result);
+        }
+    }
+}
+
+impl Drop for Responder {
+    fn drop(&mut self) {
+        if let Some(inner) = self.0.take() {
+            // The service call handler has dropped its responder without responding. This could be
+            // due to a panic or some other flaw in implementation. Reply with a generic error
+            // message.
+            inner.respond(Err("Internal server error".into()))
```

Comment:
If it could also be a logic error, I would recommend that the error message should have a little more detail, such as `Internal server error: service was dropped without sending a response`